### ISSUE TITLE
Wrong method parameter autocompletion when defined in a subclass with a constant imported

### ISSUE THREAD

**Issue by Viicos** [microsoft/pylance-release#5534]
## Environment data

-   Language Server version: Latest Pylance and prerelease
-   OS and version: Ubuntu 22
-   Python version (& distribution if applicable, e.g. Anaconda): 3.10

## Code Snippet

```python
from cmd import IDENTCHARS

class Base:

    def meth(self, arg: str):
        pass

class Impl(Base):
    def meth(self, arg: IDENTCHARS):
        return super().meth(arg)
```

## Repro Steps

When typing `def m..` in the `Impl` class, the `arg` type annotation uses the imported `PROMPT` constant instead of `str` (`IDENTCHARS` is defined as `str` in stdlib stubs, this doesn't occur for non-str constants).
[Screencast from 28-02-24 17:24:57.webm](https://github.com/microsoft/pylance-release/assets/65306057/e01ff03c-016a-45d5-a835-38a423e9097e)

**Comment by erictraut** [comment]
`IDENTCHARS` defined as _a variable_ of type `str`. Variables are not allowed in type annotations, so you cannot use `IDENTCHARS` in this manner. You should use `str` here instead ... unless your intent is to use `IDENTCHARS` as the default argument value for the `args` parameter, in which case you should switch from `arg: IDENTCHARS` to `arg = IDENTCHARS`.

**Comment by Viicos** [comment]
Sorry, the code snippet isn't really clear but the shared screencast should explain the issue better. Unfortunately, Github doesn't render `webm` files but you can open it in a new tab

**Comment by rchiodo** [comment]
Wow super weird. It's using the variable instead of 'str' in the completion.

I can repro it too. Starting code is this:

```python
from cmd import IDENTCHARS

class Base:

    def meth(self, arg: str):
        pass

class Impl(Base):
    def m
```

Then you hit tab to complete the `meth` method. And it sticks in the code that @Viicos showed.

**Comment by Viicos** [comment]
Seems to have been fixed in one of the latest versions of Pylance.

**Comment by debonte** [comment]
This looks like the same issue as https://github.com/microsoft/pylance-release/issues/5693 which was fixed in 2024.4.100.
